tests/dragonfly/connection_test.py

```diff
@@ -287,6 +287,50 @@ def generate(max):
     assert state, message
 
 
+"""
+Test that pubsub clients who are stuck on backpressure from a slow client (the one in the test doesn't read messages at all)
+will eventually unblock when it disconnects.
+"""
+
+
+@pytest.mark.asyncio
+@pytest.mark.slow
+@dfly_args({"proactor_threads": "1", "pipeline_queue_limit": "100"})
+async def test_publish_stuck(df_server: DflyInstance, async_client: aioredis.Redis):
+    reader, writer = await asyncio.open_connection("127.0.0.1", df_server.port, limit=10)
+    writer.write(b"SUBSCRIBE channel\r\n")
+    await writer.drain()
+
+    async def pub_task():
+        payload = "msg" * 1000
+        p = async_client.pipeline()
+        for _ in range(1000):
+            p.publish("channel", payload)
+        await p.execute()
+
+    publishers = [asyncio.create_task(pub_task()) for _ in range(20)]
+
+    await asyncio.sleep(5)
+
+    # Check we reached the limit
+    pub_bytes = int((await async_client.info())["dispatch_queue_pub_bytes"])
+    assert pub_bytes >= 100
+
+    await asyncio.sleep(0.1)
```

Comment:
I marked it as slow but I really suppose it's fragile because the CI might take a long time to fill the send-q/recv-q